digdag-standards/src/main/java/io/digdag/standards/operator/redshift/BaseRedshiftLoadOperator.java

```diff
@@ -0,0 +1,230 @@
+            sessionCredentialsFactory.withRoleArn(roleArn.get());
+            Optional<String> roleSessionName = config.getOptional("role_session_name", String.class);
+            if (roleSessionName.isPresent()) {
+                sessionCredentialsFactory.withRoleSessionName(roleSessionName.get());
+            }
+        }
+
+        Optional<Integer> durationSeconds = config.getOptional("session_duration", Integer.class);
+        if (durationSeconds.isPresent()) {
+            sessionCredentialsFactory.WithDurationSeconds(durationSeconds.get());
+        }
+
+        return sessionCredentialsFactory.get();
+    }
+
+    protected abstract List<AcceptableUri> buildAcceptableUriForSessionCredentials(Config config, AWSCredentials baseCredential);
+
+    protected abstract T createStatementConfig(Config params, AWSSessionCredentials sessionCredentials, String queryId);
+
+    protected abstract String buildSQLStatement(RedshiftConnection connection, T statementConfig);
+
+    protected abstract void beforeConnect(AWSCredentials credentials, T statemenetConfig);
+
+    @Override
+    protected TaskResult run(Config params, Config state, RedshiftConnectionConfig connectionConfig)
+    {
+        boolean strictTransaction = strictTransaction(params);
+
+        String statusTableName;
+        DurationParam statusTableCleanupDuration;
+        if (strictTransaction) {
+            statusTableName = params.get("status_table", String.class, "__digdag_status");
+            statusTableCleanupDuration = params.get("status_table_cleanup", DurationParam.class,
+                    DurationParam.of(Duration.ofHours(24)));
+        }
+        else {
+            statusTableName = null;
+            statusTableCleanupDuration = null;
+        }
+
+        UUID queryId;
+        // generate query id
+        if (!state.has(QUERY_ID)) {
+            // this is the first execution of this task
+            logger.debug("Generating query id for a new {} task", type());
+            queryId = UUID.randomUUID();
+            state.set(QUERY_ID, queryId);
+            throw TaskExecutionException.ofNextPolling(0, ConfigElement.copyOf(state));
+        }
+        queryId = state.get(QUERY_ID, UUID.class);
+
+        AWSCredentials baseCredentials = createBaseCredential(context.getSecrets());
+        AWSSessionCredentials sessionCredentials = createSessionCredentials(params, baseCredentials);
+        T statementConfig = createStatementConfig(params, sessionCredentials, queryId.toString());
+
+        beforeConnect(baseCredentials, statementConfig);
+
+        try (RedshiftConnection connection = connect(connectionConfig)) {
+            String query = buildSQLStatement(connection, statementConfig);
+
+            Exception statementError = connection.validateStatement(query);
+            if (statementError != null) {
+                statementConfig.accessKeyId = "********";
+                statementConfig.secretAccessKey = "********";
```

Comment:
How about moving above two lines to a new method like `StatementConfig.maskSecurityParams()`? StatementConfig.sessionToken should also be masked (if it is set).